crates/syntax/src/ast/token_ext.rs

```diff
@@ -319,6 +319,32 @@ impl ast::IntNumber {
             Some(suffix)
         }
     }
+
+    pub fn float_value(&self) -> Option<FloatTypeWrapper> {
+        let (_, text, _) = self.split_into_parts();
+        let float_value = text.parse::<f64>().ok()?;
+        Some(FloatTypeWrapper::new(float_value))
+    }
+}
+
+#[derive(Default, Debug, Clone, Eq, PartialEq)]
+pub struct FloatTypeWrapper {
+    //We convert float values into bits and that's how we don't need to deal with f32 and f64.
```

Comment:
Nit: missing space after `//` and I'd probably go for a tuple struct (`struct FloatEq(u64);`), but it doesn't matter too much.

Reply:
Hey, sorry for bringing it up here, tho I am not able to reach Jake ( @~jhgg ) for further review on this PR:
https://github.com/rust-lang/rust-analyzer/pull/12347

I try to press the re-request review button, but it doesn't do anything :(

Can you have a look at that PR too?